src/parcels/_core/particle.py

```diff
@@ -116,14 +116,27 @@ def add_variable(self, variable: Variable | list[Variable]):
         return ParticleClass(variables=self.variables + variable)
 
 
-class KernelParticle:
-    """Simple class to be used in a kernel that links a particle (on the kernel level) to a particle dataset."""
+class ParticleSetView:
```

Comment:
Is this particle.py file the best place for the `ParticleSetView` and `ParticleSetViewArray` classes? Or better to move them to particle.py; or its own file?